components/prism-scheme.js

```diff
@@ -17,7 +26,22 @@ Prism.languages.scheme = {
 		lookbehind: true
 	},
 	'number': {
-		pattern: /([\s()])[-+]?(?:\d+\/\d+|\d*\.?\d+(?:\s*[-+]\s*\d*\.?\d+i)?)\b/,
+		// This pattern (apart from the lookarounds) works like this:
+		//
+		// Decimal numbers
+		// <dec real>       := \d*\.?\d+(?:[eE][+-]?\d+)?|\d+\/\d+
+		// <dec complex>    := <dec real>(?:[+-]<dec real>i)?|<dec real>i
+		// <dec prefix>     := (?:#d(?:#[ei])?|#[ei](?:#d)?)?
+		// <dec number>     := <dec prefix>[+-]?<complex>
+		//
+		// Binary, octal, and hexadecimal numbers
+		// <b.o.x. real>    := [\da-fA-F]+(?:\/[\da-fA-F]+)?
+		// <b.o.x. complex> := <b.o.x. real>(?:[+-]<b.o.x. real>i)?|<b.o.x. real>i
+		// <b.o.x. prefix>  := #[box](?:#[ei])?|#[ei](?:#[box])?
+		// <b.o.x. number>  := <b.o.x. prefix>[+-]?<b.o.x. complex>
+		//
+		// <number>         := <dec number>|<b.o.x. number>
+		pattern: /([\s()])(?:(?:#d(?:#[ei])?|#[ei](?:#d)?)?[+-]?(?:(?:\d*\.?\d+(?:[eE][+-]?\d+)?|\d+\/\d+)(?:[+-](?:\d*\.?\d+(?:[eE][+-]?\d+)?|\d+\/\d+)i)?|(?:\d*\.?\d+(?:[eE][+-]?\d+)?|\d+\/\d+)i)|(?:#[box](?:#[ei])?|#[ei](?:#[box])?)[+-]?(?:[\da-fA-F]+(?:\/[\da-fA-F]+)?(?:[+-][\da-fA-F]+(?:\/[\da-fA-F]+)?i)?|[\da-fA-F]+(?:\/[\da-fA-F]+)?i))(?=[\s()]|$)/,
```

Comment:
Love this documentation.

Reply:
Honestly, I don't. 
It explains the pattern by showing its simpler components, yes, but how do you know that the pattern is actually what's written there? I actually manually "generated" the pattern by substituting the rules myself, so I can tell you it's a pain... especially if you change something...

I'd rather have the pattern generated from a BNF. 
We could implement a BNF-like syntax with #2190 and a syntax like this:

```js
bnf({
	// Decimal numbers
	'dec real'    : /\d*\.?\d+(?:[eE][+-]?\d+)?|\d+\/\d+/.source,
	'dec complex' : /<dec real>(?:[+-]<dec real>i)?|<dec real>i/.source,
	'dec prefix'  : /(?:#d(?:#[ei])?|#[ei](?:#d)?)?/.source,
	'dec number'  : /<dec prefix>[+-]?<complex>/.source,
	
	// Binary, octal, and hexadecimal numbers
	'box real'    : /[\da-fA-F]+(?:\/[\da-fA-F]+)?/.source,
	'box complex' : /<box real>(?:[+-]<box real>i)?|<box real>i/.source,
	'box prefix'  : /#[box](?:#[ei])?|#[ei](?:#[box])?/.source,
	'box number'  : /<box prefix>[+-]?<box complex>/.source,

	'out'         : /<dec number>|<box number>/.source,
}); // returns a RegExp
```

But this has a whole range of problems associated with it one of them being that minification basically impossible.